### PR TITLE
Field migration

### PR DESCRIPTION
**Pull request by thomasplevy**
## Description

Adds better data migration for legacy field options

## How has this been tested?

New + existing phpunit tests

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

**Comment by thomasplevy** [comment]
lifterlms_user_info_field_address_account_visibility
lifterlms_user_info_field_address_checkout_visibility
lifterlms_user_info_field_address_registration_visibility
lifterlms_user_info_field_email_confirmation_account_visibility
lifterlms_user_info_field_email_confirmation_checkout_visibility
lifterlms_user_info_field_email_confirmation_registration_visibility
lifterlms_user_info_field_names_account_visibility
lifterlms_user_info_field_names_checkout_visibility
lifterlms_user_info_field_names_registration_visibility
lifterlms_user_info_field_phone_account_visibility
lifterlms_user_info_field_phone_checkout_visibility
lifterlms_user_info_field_phone_registration_visibility

lifterlms_registration_generate_username    
lifterlms_voucher_field_registration_visibility

File: includes/forms/class-llms-form-templates.php [hunk 1]
```diff
@@ -179,6 +231,37 @@ private static function get_confirm_group_controller( $block ) {
...
+	/**
+	 * Retrieves legacy options value for a given field and location
+	 *
+	 * @since [version]
+	 *
+	 * @param string $field_id The field's identifier as found in the block schema list returned by LLMS_Form_Templates::get_reusable_block_schema().
+	 * @param string $location Form location. Accepts "checkout", "registration", or "account".
+	 * @return string
+	 */
+	private static function get_legacy_options( $field_id, $location ) {
```
Comment (eri-trabiccolo) [inline includes/forms/class-llms-form-templates.php]:
```suggestion
	private static function get_legacy_option( $field_id, $location ) {
```

I think this must be singular

File: includes/forms/class-llms-form-templates.php [hunk 2]
```diff
@@ -179,6 +231,37 @@ private static function get_confirm_group_controller( $block ) {
 
 	}
 
+	/**
+	 * Retrieves legacy options value for a given field and location
```
Comment (eri-trabiccolo) [inline includes/forms/class-llms-form-templates.php]:
```suggestion
	 * Retrieves legacy option's value for a given field and location
```

File: includes/forms/class-llms-form-templates.php [hunk 3]
```diff
@@ -121,6 +114,65 @@ private static function find_reusable_block( $field_id ) {
...
+	 * @param boolean $reusable Whether or not a reusable block should be retrieved.
+	 * @return array
+	 */
+	private static function get_block( $field_id, $location, $reusable ) {
+
+		if ( $reusable ) {
+			return self::get_reusable_block( $field_id );
+		}
+
+		$legacy_opt = self::get_legacy_options( $field_id, $location );
```
Comment (eri-trabiccolo) [inline includes/forms/class-llms-form-templates.php]:
```suggestion
		$legacy_opt = self::get_legacy_option( $field_id, $location );
```

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
Works fine, tested both a migration from 4.x and a fresh install.

Change requested is really minor, I just think the `get_legacy_options` should be named `get_legacy_option` :D

**Pull request merged**